### DISCUSSION TITLE
help installing ARM

### DISCUSSION THREAD

**Discussion by saltymedic** [automatic-ripping-machine/automatic-ripping-machine#1042] [Q&A]
Running linux server 20.04

I have attempted to installed ARM using the documentation found here:
[https://github.com/automatic-ripping-machine/automatic-ripping-machine/wiki/docker](https://github.com/automatic-ripping-machine/automatic-ripping-machine/wiki/docker)

I had difficulties understanding what to replace int he start_arm_container.sh file, but I managed to change what I could figure out. Here's the contents of what I have:
-------------------

#!/bin/bash
docker run -d \
    -p "8181:8181" \
    -e ARM_UID="1006" \
    -e ARM_GID="1006" \
    -v "/home/arm" \
    -v "/home/arm/music" \
    -v "/home/arm/logs" \
    -v "/home/arm/media" \
    -v "/etc/arm/config" \
    --device="/dev/sr0:/dev/sr0" \
    --device="/dev/sr1:/dev/sr1" \
    --device="/dev/sr2:/dev/sr2" \
    --device="/dev/sr3:/dev/sr3" \
    --privileged \
    --restart "always" \
    --name "arm-rippers" \
    --cpuset-cpus='4' \
    automaticrippingmachine/automatic-ripping-machine:latest

--------------------
I changed the port number as 8080 was being used by something else
I added the UID and GID that corresponded to the arm user
I didn't change the volume mounts as I'm just trying to get it running for now. The place where I want my videos and music to be eventually will be in a different location.
I didn't change the --device settings as I wasn't quite sure what I was looking for.
I changed the cpuset-cups to 4 as I'm not sure what I want it to look like yet

Outupt of lsscsi -g for my DVD drive entry only:

`[6:0:0:0]    cd/dvd  TSSTcorp CDDVDW SE-218CB  MF00  /dev/sr0   /dev/sg5 `

After editing these things, when I run 

`sudo nano start_arm_container.sh`

I get 

'7121ec761dc331e9bf66aaf2156ad04a79af800d7416bedb33fd333bfb246892'

I try going to the IP address for the server with the port number I designated and I get nothing.

I have very little experience with docker and I'm not sure what to do to troubleshoot

Thank you for any help

Jason

**Comment by microtechno9000** [comment]
What does "docker logs automatic-ripping-machine" output

**Reply by saltymedic** [reply]
That shows:
`Error response from daemon: No such container: automatic-ripping-machine`

However, docker logs arm-rippers shows a lot. Here's the last bit of it:


*** Killing all processes...
Jan 20 23:06:27 7121ec761dc3 syslog-ng[11]: syslog-ng shutting down; version='3.25.1'
*** Running /etc/my_init.d/10_syslog-ng.init...
Jan 20 23:06:31 7121ec761dc3 syslog-ng[11]: syslog-ng starting up; version='3.25.1'
*** Running /etc/my_init.d/arm_user_files_setup.sh...
Updating arm user id from 1000 to 1006...
usermod: no changes
Updating arm group id from 1000 to 1006...
Adding arm user to 'render' group
rm: cannot remove '/home/arm/music': Is a directory
*** /etc/my_init.d/arm_user_files_setup.sh failed with status 1

*** Killing all processes...
Jan 20 23:06:32 7121ec761dc3 syslog-ng[11]: syslog-ng shutting down; version='3.25.1'
*** Running /etc/my_init.d/10_syslog-ng.init...
Jan 20 23:06:36 7121ec761dc3 syslog-ng[11]: syslog-ng starting up; version='3.25.1'
*** Running /etc/my_init.d/arm_user_files_setup.sh...
Updating arm user id from 1000 to 1006...
usermod: no changes
Updating arm group id from 1000 to 1006...
Adding arm user to 'render' group
rm: cannot remove '/home/arm/music': Is a directory
*** /etc/my_init.d/arm_user_files_setup.sh failed with status 1

*** Killing all processes...
Jan 20 23:06:37 7121ec761dc3 syslog-ng[11]: syslog-ng shutting down; version='3.25.1'
*** Running /etc/my_init.d/10_syslog-ng.init...
Jan 20 23:06:41 7121ec761dc3 syslog-ng[12]: syslog-ng starting up; version='3.25.1'
*** Running /etc/my_init.d/arm_user_files_setup.sh...
Updating arm user id from 1000 to 1006...
usermod: no changes
Updating arm group id from 1000 to 1006...
Adding arm user to 'render' group
rm: cannot remove '/home/arm/music': Is a directory
*** /etc/my_init.d/arm_user_files_setup.sh failed with status 1

*** Killing all processes...
Jan 20 23:06:42 7121ec761dc3 syslog-ng[12]: syslog-ng shutting down; version='3.25.1'
*** Running /etc/my_init.d/10_syslog-ng.init...
Jan 20 23:06:46 7121ec761dc3 syslog-ng[11]: syslog-ng starting up; version='3.25.1'
*** Running /etc/my_init.d/arm_user_files_setup.sh...
Updating arm user id from 1000 to 1006...
usermod: no changes
Updating arm group id from 1000 to 1006...
Adding arm user to 'render' group
rm: cannot remove '/home/arm/music': Is a directory
*** /etc/my_init.d/arm_user_files_setup.sh failed with status 1

*** Killing all processes...
Jan 20 23:06:47 7121ec761dc3 syslog-ng[11]: syslog-ng shutting down; version='3.25.1'
*** Running /etc/my_init.d/10_syslog-ng.init...
Jan 20 23:06:53 7121ec761dc3 syslog-ng[11]: syslog-ng starting up; version='3.25.1'
*** Running /etc/my_init.d/arm_user_files_setup.sh...
Updating arm user id from 1000 to 1006...
usermod: no changes
Updating arm group id from 1000 to 1006...
Adding arm user to 'render' group
rm: cannot remove '/home/arm/music': Is a directory
*** /etc/my_init.d/arm_user_files_setup.sh failed with status 1

*** Killing all processes...
Jan 20 23:06:54 7121ec761dc3 syslog-ng[11]: syslog-ng shutting down; version='3.25.1'
*** Running /etc/my_init.d/10_syslog-ng.init...
Jan 20 23:07:04 7121ec761dc3 syslog-ng[11]: syslog-ng starting up; version='3.25.1'
*** Running /etc/my_init.d/arm_user_files_setup.sh...
Updating arm user id from 1000 to 1006...
usermod: no changes
Updating arm group id from 1000 to 1006...
Adding arm user to 'render' group
rm: cannot remove '/home/arm/music': Is a directory
*** /etc/my_init.d/arm_user_files_setup.sh failed with status 1

*** Killing all processes...
Jan 20 23:07:05 7121ec761dc3 syslog-ng[11]: syslog-ng shutting down; version='3.25.1'
*** Running /etc/my_init.d/10_syslog-ng.init...
Jan 20 23:07:20 7121ec761dc3 syslog-ng[11]: syslog-ng starting up; version='3.25.1'
*** Running /etc/my_init.d/arm_user_files_setup.sh...
Updating arm user id from 1000 to 1006...
usermod: no changes
Updating arm group id from 1000 to 1006...
Adding arm user to 'render' group
rm: cannot remove '/home/arm/music': Is a directory
*** /etc/my_init.d/arm_user_files_setup.sh failed with status 1

*** Killing all processes...
Jan 20 23:07:21 7121ec761dc3 syslog-ng[11]: syslog-ng shutting down; version='3.25.1'
*** Running /etc/my_init.d/10_syslog-ng.init...
Jan 20 23:07:49 7121ec761dc3 syslog-ng[11]: syslog-ng starting up; version='3.25.1'
*** Running /etc/my_init.d/arm_user_files_setup.sh...
Updating arm user id from 1000 to 1006...
usermod: no changes
Updating arm group id from 1000 to 1006...
Adding arm user to 'render' group
rm: cannot remove '/home/arm/music': Is a directory
*** /etc/my_init.d/arm_user_files_setup.sh failed with status 1

*** Killing all processes...
Jan 20 23:07:50 7121ec761dc3 syslog-ng[11]: syslog-ng shutting down; version='3.25.1'
*** Running /etc/my_init.d/10_syslog-ng.init...
Jan 20 23:08:44 7121ec761dc3 syslog-ng[11]: syslog-ng starting up; version='3.25.1'
*** Running /etc/my_init.d/arm_user_files_setup.sh...
Updating arm user id from 1000 to 1006...
usermod: no changes
Updating arm group id from 1000 to 1006...
Adding arm user to 'render' group
rm: cannot remove '/home/arm/music': Is a directory
*** /etc/my_init.d/arm_user_files_setup.sh failed with status

**Reply by microtechno9000** [reply]
That is an issue a few users have encountered, its being worked on at the moment to resolve.

A work around is to remove the music folder in /home/arm, or if it is a symbolic link then remove the link

**Reply by saltymedic** [reply]
The music directory never gets created. The docker startup file tells it that the directory is located in the /home/arm directory

The contents of the  /home/arm directory are as follows:

drwxr-xr-x 3 arm  arm  4096 Jan 20 12:34 ./
drwxr-xr-x 9 root root 4096 Jan 15 10:41 ../
-rwxr-xr-x 1 root root  671 Jan 15 10:50 backup-start_arm_container.sh*
-rw-r--r-- 1 arm  arm   220 Apr  8  2014 .bash_logout
-rw-r--r-- 1 arm  arm  3771 May 16  2017 .bashrc
drwxr-xr-x 5 arm  arm  4096 Jan 15 10:54 .config/
-rw-r--r-- 1 arm  arm   807 May  3  2019 .profile
-rwxrwxr-x 1 arm  arm   521 Jan 20 11:10 start_arm_container.sh*
-rw-r--r-- 1 arm  arm  1600 Jan  9  2016 .Xdefaults
-rw-r--r-- 1 arm  arm    14 Jan  9  2016 .xscreensaver

**Reply by microtechno9000** [reply]
Ok
Your docker run command is missing a local folder for docker to host files. the command should have two locations, a local directory outside docker and the docker reference.
details here https://docs.docker.com/engine/reference/commandline/container_run/#volume
